crates/wasmtime/src/func.rs

```diff
@@ -1559,12 +1423,29 @@ pub trait IntoFunc<Params, Results> {
 /// be removed in the future at some point after interface types is implemented. If
 /// you're relying on this Caller type it's recommended to become familiar with
 /// interface types to ensure that your use case is covered by the proposal.
-pub struct Caller<'a> {
-    store: &'a Store,
-    caller_vmctx: *mut VMContext,
+pub struct Caller<'a, T> {
+    pub(crate) store: StoreContextMut<'a, T>,
+    caller: &'a InstanceHandle,
 }
 
-impl Caller<'_> {
+impl<T> Caller<'_, T> {
+    unsafe fn with<R>(caller: *mut VMContext, f: impl FnOnce(Caller<'_, T>) -> R) -> R {
```

Comment:
Previously this function would have used `with_last_info` from the `traphandlers` module, but now it looks up the store from within the provided `caller` `*mut VMContext`. Note that previously the pointer could be null but I've now had it arranged such that this pointer is never null.